pkg/vm/memory/memory_manager.go

```diff
@@ -8,14 +8,45 @@ type MemoryManager struct {
 	Memory *Memory
 }
 
-func CreateMemoryManager() (*MemoryManager, error) {
+// Creates a new memory manager
+func CreateMemoryManager() *MemoryManager {
 	memory := InitializeEmptyMemory()
 
 	return &MemoryManager{
 		Memory: memory,
-	}, nil
+	}
 }
 
-func (mm *MemoryManager) GetByteCodeAt(segmentIndex uint64, offset uint64) *f.Element {
-	return nil
+// It returns all segments in memory but relocated as a single segment
+// Each element is a pointer to a field element, if the cell was not accessed,
+// nil is stored instead
+func (mm *MemoryManager) RelocateMemory() []*f.Element {
+	maxMemoryUsed := 0
+	// segmentsOffsets[0] =  0
+	// segmentsOffsets[1] = len(segment[0])
+	// segmentsOffsets[N] = len(segment[n - 1]) + sum of segmentsOffsets[n - i] for i in [0, n-1]
+	segmentsOffsets := make([]int, len(mm.Memory.Segments)+1)
```

Comment:
What's the advantage of making the offsets slice ```[]int``` not ```[]uint64```?

Reply:
None, changed it to `[]uint64`